src/wallet.js

```diff
@@ -134,12 +134,30 @@ const wallet = {
       pbkdf2Hasher: 'sha1', // For now we are only using SHA1
     }
 
+    return this.startWallet(access, privkey.xpubkey, loadHistory);
+  },
+
+  /**
+   * Set wallet data on storage and start it
+   *
+   * @param {Object} accessData Object of data to be saved on storage. Will be empty for hardware wallet and for software will have the keys
+   *                    (mainKey, hash, salt, words, hashPasswd, saltPasswd, hashIterations, pbkdf2Hasher) as set on executeGenerateWallet method
+   * @param {string} xpub Xpub string of the wallet being started
+   * @param {boolean} loadHistory if should load the history from the generated addresses
+   *
+   * @return {Promise} Promise that resolves when finishes loading address history, in case loadHistory = true, else returns null
+   * @memberof Wallet
+   * @inner
+   */
+  startWallet(accessData, xpub, loadHistory) {
+    WebSocketHandler.setup();
+    this.afterOpen();
```

Comment:
Shouldn't we call it after the connection has been established?

Reply:
This does not refer to the websocket being open, but to the wallet state, as I explained on the other comment.

I change the method name to `setWalletAsOpen`.

Reply:
Great. Thanks!